src/microsim/MSVehicle.cpp

```diff
@@ -4313,6 +4313,9 @@ MSVehicle::executeMove() {
             // check (#2681): Can this be skipped?
             myLaneChangeModel->prepareStep();
         } else {
+            if (MSGlobals::gLaneChangeDuration > 0 && !myLaneChangeModel->isChangingLanes() && !(myLaneChangeModel->getSigmaLat() > 0)) {
```

Comment:
I don't like leaking thos lcmodel details into MSVehicle. Seem though about putting this only into the alternative branch to  to prepareStep (i'd rather call it unconditionally and have prepareStep do changes on top if needed.

Reply:
The new code is missing `!(myLaneChangeModel->getSigmaLat() > 0)`. As I wrote in the other comment, `setSpeedLat` should only be called once. The new function is a bit cleaner and seems to (almost) do the same as the Code here and below.

Reply:
My idea was to have the sigmaLat stuff be handled *afterwards* in `prepare step` (so we don't have to worry about the details here). `prepareStep` computes a completely new speedLat anyway when using sigmaLat.